bson/bsonrw/extjson_reader.go

```diff
@@ -164,6 +164,19 @@ func (ejvr *extJSONValueReader) skipObject() {
 	depth := 1
 	for depth > 0 {
 		ejvr.p.advanceState()
+
+		// If object is empty, raise depth and return. If there is a comma, there are
+		// remaining fields, emptyObject must be set back to false, and comma must be
+		// skipped with advanceState().
+		if ejvr.p.emptyObject {
+			if ejvr.p.s == jpsSawComma {
+				ejvr.p.emptyObject = false
+				ejvr.p.advanceState()
+			}
+			depth--
+			return
```

Comment:
Do we want to `return` or `continue`? what happens if we need to skip something with a nested empty document, something like
`"UndefinedField": {"empty": {}, "key": 1}`

Reply:
`continue` is more correct, but it actually doesn't change the parsing behavior 💭 

In that example, the parser will realize `UndefinedField` is undefined in the destination struct and will call `skipObject` on the value of `UndefinedField`: `{"empty": {}, "key": 1}`. Since that value is not an empty object itself, the `emptyObject` flag will not have been set, so the logic in this block won't actually apply. That said, `continue`ing is more accurate than `return`ing in my opinion, since the `depth` variable will be set back to `0` and return anyway.

Also updated the existing test for `"empty embedded document"` to be the case you described above.